flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosTaskManagerParameters.java

```diff
@@ -419,6 +411,34 @@ public static MesosTaskManagerParameters create(Configuration flinkConfig) {
 			uris);
 	}
 
+	private static ContaineredTaskManagerParameters createContaineredTaskManagerParameters (final Configuration flinkConfig) {
+		double cpus = getCpuCores(flinkConfig);
+		MemorySize totalProcessMemory = MemorySize.parse(flinkConfig.getInteger(MESOS_RM_TASKS_MEMORY_MB) + "m");
+		TaskExecutorResourceSpec taskExecutorResourceSpec = TaskExecutorResourceUtils
+			.newResourceSpecBuilder(flinkConfig)
+			.withCpuCores(cpus)
+			.withTotalProcessMemory(totalProcessMemory)
+			.build();
+
+		return ContaineredTaskManagerParameters.create(flinkConfig, taskExecutorResourceSpec);
+	}
+
+	private static double getCpuCores(final Configuration configuration) {
```

Comment:
I think we can unify this function in `TaskExecutorResourceUtils#getCpuCores` adding a nullable optional option, to fallback to (for yarn and mesos but none for standalone), before falling back to fraction/number of slots.

Reply:
I would try to avoid adding any yarn / mesos specific codes in `TaskExecutorResourceUtils`.
- I'm in favor of keeping `TaskExecutorResourceUtils` for general common purpose only.
- Parsing mesos / yarn config options will introduce extra dependencies on flink-yarn / flink-mesos into flink-runtime.

Reply:
If the method accepts a general option to fallback to, then there is no dependency on yarn/mesos in `TaskExecutorResourceUtils`.